stixcore/products/tests/test_bsd.py

```diff
@@ -0,0 +1,27 @@
+import re
+from unittest.mock import patch
+
+import pytest
+
+from stixcore.data.test import test_data
+from stixcore.products.level0.science import CompressedPixelData
+
+testpackets = [(test_data.tmtc.TM_21_6_21_complete, CompressedPixelData, 'xray-cpd')]
+
+
+@patch('stixcore.products.levelb.binary.LevelB')
+@pytest.mark.parametrize('packets', testpackets, ids=[f[0].name for f in testpackets])
+def test_xray(levelb, packets):
+    hex_file, cl, name = packets
+    with hex_file.open('r') as file:
+        hex = file.readlines()
+
+    levelb.data.__getitem__.return_value = [re.sub(r"\s+", "", h) for h in hex]
+
+    xray_L0 = cl.from_levelb(levelb)
```

Comment:
So this bypassed the code which generates and check for a complete packet sequence passing it incomplete sequence will give undefined behaviour. I'm not sure where is best to put this kind of check I think maybe we can expand and subclass `PacketSequence` or add `CompletePacketSequence` class?